api/gateway/share.go

```diff
@@ -93,33 +91,14 @@ func (h *Handler) handleDataByNamespaceRequest(w http.ResponseWriter, r *http.Re
 	}
 }
 
-func (h *Handler) getShares(ctx context.Context, height uint64, nID namespace.ID) ([]share.Share, int64, error) {
-	// get header
-	var (
-		err    error
-		header *header.ExtendedHeader
-	)
-
-	//TODO: change this to NetworkHead once the adjacency in the store is fixed.
-	header, err = h.header.LocalHead(ctx)
+func (h *Handler) getShares(ctx context.Context, height uint64, nID namespace.ID) ([]share.Share, int, error) {
+	header, code, err := headerGetByHeight(ctx, int(height), h.header)
 	if err != nil {
-		return nil, 0, err
+		return nil, code, err
 	}
 
-	if height > 0 {
-		if storeHeight := uint64(header.Height()); storeHeight < height {
-			return nil, 0, fmt.Errorf(
-				"current head local chain head: %d is lower than requested height: %d"+
-					" give header sync some time and retry later", storeHeight, height)
-		}
-		header, err = h.header.GetByHeight(ctx, height)
-	}
-	if err != nil {
-		return nil, 0, err
-	}
-	// perform request
 	shares, err := h.share.GetSharesByNamespace(ctx, header.DAH, nID)
-	return shares.Flatten(), header.Height(), err
+	return shares.Flatten(), http.StatusInternalServerError, err
```

Comment:
if `GetSharesByNamespace` fails, you can get `nil` instead of shares and it will panic

Reply:
True, so the back was there before